packages/cojson/src/sync.ts

```diff
@@ -680,10 +618,6 @@ export class SyncManager {
             );
         }
 
-        if (resolveAfterDone) {
-            resolveAfterDone(coValue);
-        }
```

Comment:
Moved the resolve up.

Promises are resolved on the next tick of the event loop anyway so, since there are no await in the middle, it shouldn't matter where we resolve the covalue state.